pkg/dynamicinstrumentation/module.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package dynamicinstrumentation
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+)
+
+type Module struct{}
+
+func NewModule(config *Config) (*Module, error) {
```

Comment:
Linter errors are right - you have an exclusion flag in the other file but no build flag here so for non-Linux platforms both files will be built, causing name collision. I would probably suggest that you maybe name this file `module_linux` just to make it clear from that standpoint that this only works on that platform.